### ISSUE TITLE
ic-text-field, ic-select: Reflow problems

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-ui-kit#4071]
### Summary of the bug 
At narrow viewports, ic-text-field and ic-select overflows the screen. Minor reflow failure. It seems that we've set the fields to have a minimum width of 320px, which would be conformant if apps don't provide any padding around them at all.

This issue presents itself in our guidance site and storybook, it seems to be the width CSS rule on `ic-input-component-container`. Turn the rule off and the fields shrink happily (but also grow to fill the width available, so some more work is required)

### 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to [React ic-text-field storybook](https://mi6.github.io/ic-ui-kit/branches/develop/react/?path=/story/react-components-text-field--default)
2. Shrink viewport to 320px horizontal using dev tools
3. See how the field spills off the side of the screen. 

### 📸 Screenshots or code 

Evidence of issue happening in guidance site and storybook, and the line of CSS that seems to be the issue

<img width="459" height="369" alt="Image" src="https://github.com/user-attachments/assets/f0e98e6c-1741-4ab8-ac3f-bee820cd3952" />

<img width="267" height="194" alt="Image" src="https://github.com/user-attachments/assets/3b762171-abfb-425e-a802-d2b2984f3eca" />

<img width="349" height="351" alt="Image" src="https://github.com/user-attachments/assets/d2cfd7c1-370a-4615-a6eb-87d9217e7fbb" />

### 🧐 Expected behaviour 
ic-text-field should shrink to remain within screen bounds. 

### 🚨 Urgency (low, medium or high)
If applicable, tell us how urgent it is that this issue gets resolved, based on the impact it has on your team's work or project timeline.

High

### Additional info 
This will help address https://github.com/mi6/ic-design-system/issues/1706
